lib/tokenize.cpp

```diff
@@ -2302,7 +2302,7 @@ bool Tokenizer::simplifyUsing()
                 Token::Match(tok->linkAt(2), "] ] = ::| %name%")))))
             continue;
 
-        std::string name = tok->strAt(1);
+        const std::string& name = tok->strAt(1);
```

Comment:
This and most (all?) of the other changes seem a bit dangerous. We need to be sure that the owner is `const` and/or only `const` member functions are being called. But this holds only true for the moment. If the code is changed afterwards that would actually issues.

It is true that the code could have been written like this in the first place but that was either intentional or erroneous by the coder and not suggested by a tool perceived as "trustworthy".

I did such overzealous changes in #4034 which lead to crashes in the tests.

The example in the ticket which inspired this is valid but there is only a single usage and the call there could be inlined in the single usage and not variable is even necessary.

Reply:
I agree that this feels a bit dangerous. Tokens are deleted in this function.
Are you sure there is no path where the `tok` will be deleted before name is accessed?

Reply:
`name` is only read at L2367, deletions start at L2386. So this should be safe.

Reply:
But would cppcheck detect the issue if the code was changed in the future to read `name` after the token was deleted?

At least using a copy this will be more resilient to changes in the future.

Reply:
No, we would not detect that, see the suppression that I added elsewhere. I don't think the checker has many clues here to decide if it should warn or not.